homeassistant/components/ecowitt/__init__.py

```diff
@@ -0,0 +1,45 @@
+"""The Ecowitt Weather Station Component."""
+from __future__ import annotations
+
+from aioecowitt import EcoWittListener
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_PORT, EVENT_HOMEASSISTANT_STOP, Platform
+from homeassistant.core import Event, HomeAssistant
+
+from .const import CONF_PATH, DOMAIN
+
+PLATFORMS: list[Platform] = [Platform.SENSOR, Platform.BINARY_SENSOR]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up the Ecowitt component from UI."""
+    hass.data.setdefault(DOMAIN, {})
+
+    ecowitt = hass.data[DOMAIN][entry.entry_id] = EcoWittListener(
+        port=entry.data[CONF_PORT], path=entry.data[CONF_PATH]
+    )
+
+    hass.config_entries.async_setup_platforms(entry, PLATFORMS)
```

Comment:
This one is old. You should use 
```suggestion
    await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)
```

I've updated the scaffold script in #77429

Reply:
Also, you should start ecowitt first before setting up the platforms or else the platforms might hit an unstarted ecowitt API.